workflow-manager/kubernetes/kubernetes.go

```diff
@@ -25,6 +30,29 @@ type Client struct {
 // it is not empty. If dryRun is true, then any destructive API calls will be
 // made with DryRun: All.
 func NewClient(namespace string, kubeconfigPath string, dryRun bool) (*Client, error) {
+	// Making sure endpoint is available before returning kubernetes client
+	// only for inCluster config
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	if kubeconfigPath == "" && host != "" && port != "" {
+		available := false
+		strategy := retry.LimitTime(60*time.Second,
+			retry.Exponential{
+				Initial: 500 * time.Millisecond,
+				Factor:  1.2,
+			},
+		)
+		timeout := time.Duration(1 * time.Second)
+		for a := retry.Start(strategy, nil); a.Next(); {
+			if _, err := net.DialTimeout("tcp", host+":"+port, timeout); err == nil {
+				available = true
+				break
+			}
+		}
+		if !available {
+			panic("InCluster k8s api endpoint not available")
```

Comment:
This should return an error, so that it gets propagated up to `main()` and metrics can be incremented appropriately as in other failure modes.

Reply:
Do we have an existing error that I can throw here or do you want to create a new one?